apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts

```diff
@@ -4329,10 +4353,49 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
                 });
             }
             this.totalAdvanceReceiptsAdjustedAmount = totalAmount;
+            this.adjustPaymentData.totalAdjustedAmount = this.totalAdvanceReceiptsAdjustedAmount;
+            if (this.adjustPaymentData.totalAdjustedAmount > 0) {
+                this.isAdjustAmount = true;
+            }
         } else {
             this.advanceReceiptAdjustmentData = null;
         }
     }
+
+    /**
+     * To calculate balance due amount after adjustment of advance receipts
+     *
+     * @returns {number} Balance due amount
+     * @memberof ProformaInvoiceComponent
+     */
+    public getCalculatedBalanceDueAfterAdvanceReceiptsAdjustment(): number {
+        return this.invFormData.voucherDetails.grandTotal - this.adjustPaymentData.totalAdjustedAmount - this.depositAmount + this.invFormData.voucherDetails.tcsTotal - this.invFormData.voucherDetails.tdsTotal
+    }
+
+    /**
+     * Call API to get all advance receipts of an invoice
+     *
+     * @param {*} customerUniquename Selected customer unique name
+     * @param {*} voucherDate  Voucher Date (DD-MM-YYYY) of selected invoice
+     * @memberof ProformaInvoiceComponent
+     */
+    public getAllAdvanceReceipts(customerUniquename, voucherDate): void {
+        if (customerUniquename && voucherDate) {
+            let requestObject = {
+                accountUniqueName: customerUniquename,
+                invoiceDate: voucherDate
+            };
+            this.salesService.getAllAdvanceReceiptVoucher(requestObject).subscribe(res => {
+                if (res && res.status === 'success') {
```

Comment:
Else of this if block should be there to reset `isAccountHaveAdvanceReceipts` to `false`